### PR TITLE
change qrs and pets view

### PR DESCRIPTION
**Pull request by jOR13**
views are  not finished yet (working on it)

@itsmeurbi

### CODE REVIEW

File: app/controllers/qrs_controller.rb [hunk 1]
```diff
@@ -1,6 +1,13 @@
 class QrsController < ApplicationController
+
+  before_action :authenticate_user!
   before_action :set_qr, only: %i[ show edit update destroy ]
 
+  def public
```
Comment (itsmeurbi) [inline app/controllers/qrs_controller.rb:6]: Your actions should only be `new`, `create`, `update`, `delete` or `new`

File: app/services/qr_generator.rb [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
-    def initialize(text)
-        @text = text
+    def initialize(path, code)
+        @path = path
+        @code = code
     end
         
     def generate
-        qrcode = RQRCode::QRCode.new(@text)
+        puts "Generating QR code for #{@path}"
```
Comment (itsmeurbi) [inline app/services/qr_generator.rb:10]: Is there a reason why you need this?

**Review by itsmeurbi — CHANGES_REQUESTED** [review]
I'm a bit out of context on what your application does.
Does it needs to handle json requests? If not, let's get rid of `format.json` blocks

**Review by itsmeurbi — CHANGES_REQUESTED** [review]
Ok now this PR is HUGE!
Split it into small PRs so we can review it.

**Comment by itsmeurbi** [comment]
@jOR13 why did we merge this without any approval? 🚨

File: app/views/pets/index.html.erb [hunk 3]
```diff
@@ -1,45 +1,39 @@
...
-    'button' %>
+    <div>
+      <%= link_to "change view 👓️", pets_path(view_format: "card"), class:
+    'button is-info is-rounded' %>
+    <%= link_to "New pet", new_pet_path, class: "button is-info is-rounded" %>
+    </div>
   </div>
-
-  <div class="columns is-desktop">
+  <div class="columns is-multiline" id="pets" style="justify-content: center;">
```
[the remark this answers is not in the document]
Comment (jOR13) [reply app/views/pets/index.html.erb]: already fix it

File: app/views/pets/index.html.erb [hunk 4]
```diff
@@ -1,45 +1,39 @@
...
     <% @pets.each do |pet| %>
-    <div class="card column" style="margin: 5px">
+      <div>
+          <div class="column" id="<%= dom_id pet %>">
+    <div class="card"  style="min-height: 230px; min-width: 180px; min-height: 400px; border-radius: 20px;">
```
[the remark this answers is not in the document]
Comment (jOR13) [reply app/views/pets/index.html.erb:17]: already fix it

File: app/views/pets/index.html.erb [hunk 5]
```diff
@@ -1,45 +1,39 @@
...
       <div class="card-image">
-        <figure class="image is-4by3">
-          <%= image_tag url_for(pet.image.variant(resize_to_fill: [200, 200])),
-          class: '' if pet.image.attached? %>
+        <figure class="image is-square">
+          <%= image_tag url_for(pet.image.variant(resize_to_fit: [200, 200])),
```
[the remark this answers is not in the document]
Comment (jOR13) [reply app/views/pets/index.html.erb:20]: already fix it

File: app/views/pets/index.html.erb [hunk 6]
```diff
@@ -1,45 +1,39 @@
...
+      <div>
+          <div class="column" id="<%= dom_id pet %>">
+    <div class="card"  style="min-height: 230px; min-width: 180px; min-height: 400px; border-radius: 20px;">
       <div class="card-image">
-        <figure class="image is-4by3">
-          <%= image_tag url_for(pet.image.variant(resize_to_fill: [200, 200])),
-          class: '' if pet.image.attached? %>
+        <figure class="image is-square">
+          <%= image_tag url_for(pet.image.variant(resize_to_fit: [200, 200])), 
+          style: 'border-top-right-radius: 20px;
```
[the remark this answers is not in the document]
Comment (jOR13) [reply app/views/pets/index.html.erb:21]: already fix it

File: app/views/qrs/_qr.html.erb [hunk 7]
```diff
@@ -1,25 +1,23 @@
...
+      <figure class="image is-square">
+      <img class="is-rounded" src="/qrs/<%=qr.code%>.png" />
+    </figure>
+    </div>
+  <div class="card-content">
+    <p class="subtitle is-6" style="text-align: center"><%= qr.code %></p>
+    <div class="media">
+      <div class="media-content" style="text-align: center">
+        <% if qr.pet.name?  %>
+        <%# <%= image_tag url_for(qr.pet.image.variant(resize_to_fill: [200, 200])), class: "image is-64x64", style: "border-radius: 40px; display: flex;"%>
```
[the remark this answers is not in the document]
Comment (jOR13) [reply app/views/qrs/_qr.html.erb:13]: already fix it

File: app/views/qrs/index.html.erb [hunk 8]
```diff
@@ -1,14 +1,13 @@
...
-    </p>
-  <% end %>
+<div class="container">
+   <div style="display: flex; flex-direction: row; justify-content: space-between">
+      <h1 class="title">Qrs</h1>
+      <%= link_to "New qr", new_qr_path, class: "button is-info is-rounded" %>
+    </div>
+  <div class="columns is-multiline" id="qrs" style="justify-content: center;">
+    <% @qrs.each do |qr| %>
+    <div>
```
[the remark this answers is not in the document]
Comment (jOR13) [reply app/views/qrs/index.html.erb:11]: already fix it

File: app/controllers/qrs_controller.rb (hunk 1 above)
Comment (jOR13) [reply app/controllers/qrs_controller.rb:26]: already fix it i have a better idea how routes it works

File: app/services/qr_generator.rb (hunk 2 above)
Comment (jOR13) [reply app/services/qr_generator.rb]: na that was just for test the output, but i delete it

**Pull request merged**